### DISCUSSION TITLE
TuringCourses/major/data_structure/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#107] [General]
# TuringCourses/major/data_structure/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major/data_structure/

**Comment by MTDickens** [comment]
FDS 课程按以往惯例会有一个“诚信测验”。该测验要求必须满分，而且有些题目比较怪。

因此，为方便大家作答，这里贴一份陈越姥姥的[官方答案](https://www.zhihu.com/question/68235594/answer/261395370)，供大家参考：

https://www.zhihu.com/question/68235594/answer/261395370

**Reply by MTDickens** [reply]
由于官方答案里没有对应的题号，我在这里就同时给出：

- 答案为 F: 11, 15, 20, 21, 26, 27, 30, 52
    - 一共 8 道题为 F
- 答案为 T: （其他所有题目）

另外，鉴于题号可能会在将来发生变化，还希望大家不要直接抄这里的题号。为保险起见，还是要结合官方答案作答。

**Update**

ADS 的答案相比 DS 有变化：

- 答案为 F: 11, 15, 20, 21, 30, 31, 34, 56
    - 一共 8 道题为 F
- 答案为 T: （其他所有题目）

**Comment by Fairicle** [comment]
关于 hqm 老师的课前签到：理论上他不应该有小测，但是 22 级他总共开了四次课上小测。不过他本人也没有说这个分数的占比，第一节课的 PPT 中分数组成部分也没有小测的占比。出分之后根据身边统计学的感觉是最后没怎么占分数。

关于 project：hard 和 normal 是通过一场所谓“钻石段位分级考试”决定的，具体形式是去紫金港机房在 PTA 上面考试（全是编程题，22 级是 7 道编程题，难度从易到难有梯度），前 30% 的同学能够选择 hard（如果不想做 hard 应该和老师说就能换成 normal），剩下的同学就是做 normal。这场考试会安排在开学第一周，需要注意的是参加考试要提前在 PTA 上面报名（不过忘记报名直接跟老师讲应该也能去考，可能会麻烦一些）。hard 的难度会比 normal 高不少，花费时间更多。

由于 hard 多出来的 5 分能够溢出到总评，一般来说至少能够给总评增加三四分，还是比较划算的。不过值得注意的是在 22 级考完期末考试之后所有老师曾一度联合宣布 hard 多出来的分数不溢出到总评，不过由于学生们反应强烈最后还是算入了总评。23 级是否会出现类似情况是未知数，请大家一定要在**一开始**的时候就搞清楚是否能够溢出到总评（如果不能溢出到总评，选择 hard 反而会徒增痛苦，因为在 bonus 存在的情况下，只要不摆烂大家平时分基本都能满，hard 多的五分没什么用）。

**Comment by 626lt** [comment]
yzq老师相对遵守课程组的规定，但是有问题的地方也可以与他积极协商来得到解决，在22级上课最后hard分数能否溢出到总评的问题以及之前project延期的问题上，老师都给出了很好的回应。yzq老师的讲课也还可以，而且突击小测的占比不低，对于基础一般的同学不建议翘课。

**Comment by Whning0513** [comment]
陈超超老师project基本是给满的，bonus也很不错
